### PR TITLE
Fix: Bucket Handling

### PR DESCRIPTION
**Pull request by HenryLoenwind**
* Restored Reservoir special multi-block handling (thx git)
* Restored sneaking behavior
* re #2309

### CODE REVIEW

**Comment by tterrag1098** [comment]
I already fixed this, no?

**Comment by HenryLoenwind** [comment]
Wasn't linked in...

Quick look says it might work, but leaves other parts of the code hanging. Sneaking you missed (only tested with buckets, I'd guess?).

**Comment by tterrag1098** [comment]
I don't see why sneaking should prevent bucket filling. Did it before?

**Comment by HenryLoenwind** [comment]
On some machines it dis, on some it didn't. That was part of the reason to consolidate the code.

Sneaking is the universal "right-click inverter". With all tanks I checked this means it allows source blocks to be picked up or placed into the world. Vanilla implements this in the bucket's code, all other mod's liquid containers leave it to the block.

File: src/main/java/crazypants/enderio/BlockEio.java [hunk 1]
```diff
@@ -95,6 +95,10 @@ public boolean onBlockActivated(World world, int x, int y, int z, EntityPlayer e
       }
     }
     
+    if (entityPlayer.isSneaking()) {
```
Comment (tterrag1098) [inline src/main/java/crazypants/enderio/BlockEio.java]: I think this check should be integrated into the next if clause (i.e. `if (!entityPlayer.isSneaking() && te instanceof ITankAccess)`) so that it does not interfere with later code that might want to function if the player is sneaking.

File: src/main/java/crazypants/enderio/machine/reservoir/BlockReservoir.java [hunk 2]
```diff
@@ -69,11 +70,18 @@ public boolean onBlockActivated(World world, int x, int y, int z, EntityPlayer e
         }
         return true;
       }
+      if (FluidUtil.fillInternalTankFromPlayerHandItem(world, x, y, z, entityPlayer, tank)) {
```
Comment (tterrag1098) [inline src/main/java/crazypants/enderio/machine/reservoir/BlockReservoir.java]: Why is this necessary?

Comment (HenryLoenwind) [inline src/main/java/crazypants/enderio/machine/reservoir/BlockReservoir.java]: because the te *here* is the multi-block controller.

Comment (tterrag1098) [inline src/main/java/crazypants/enderio/machine/reservoir/BlockReservoir.java]: But there is already code for that, see getInputTank and getOutputTanks in TileReservoir

File: src/main/java/crazypants/enderio/BlockEio.java (hunk 1 above)
Comment (HenryLoenwind) [inline src/main/java/crazypants/enderio/BlockEio.java]: I did not want to write

    if (!entityPlayer.isSneaking()) {
        code(1)
    }
    if (entityPlayer.isSneaking()) {
        code(2);
    }
    code3();

and I didn't want to put the remainder of the method into an "else" bacause that would change indent-level and I know, you don't like that in commits.

File: src/main/java/crazypants/enderio/machine/reservoir/BlockReservoir.java (hunk 2 above)
Comment (HenryLoenwind) [inline src/main/java/crazypants/enderio/machine/reservoir/BlockReservoir.java]: yes, as I wrote, you did that in a way I did not see.

And btw, your code leaves setDirty() hanging in the air now.

Comment (tterrag1098) [inline src/main/java/crazypants/enderio/machine/reservoir/BlockReservoir.java]: How so?

Comment (HenryLoenwind) [inline src/main/java/crazypants/enderio/machine/reservoir/BlockReservoir.java]: it does not contain the getController() code. So it can only fix the internal tanks if the player clicks the controller block by chance. And, because you present the tanks in the wrong order, in the other 3 cases, no water regens.
